x-pack/plugins/index_management/public/application/components/mappings_editor/lib/serializers.ts

```diff
@@ -45,16 +45,19 @@ const runParametersDeserializers = (field: Field): Field =>
   );
 
 export const fieldSerializer: SerializerFunc<Field> = (field: Field) => {
+  const { otherTypeJson, ...rest } = field;
+  const updatedField: Field = Boolean(otherTypeJson) ? { ...otherTypeJson, ...rest } : { ...rest };
+
   // If a subType is present, use it as type for ES
-  if ({}.hasOwnProperty.call(field, 'subType')) {
-    field.type = field.subType as DataType;
-    delete field.subType;
+  if ({}.hasOwnProperty.call(updatedField, 'subType')) {
```

Comment:
Just curious, what is the reason we use `{}.hasOwnProperty.call` pattern here? Do we use prototypical inheritance on the field objects somewhere?

Reply:
I just use this pattern everywhere 😄 It became a habit.